### DISCUSSION TITLE
How to run Multiple Nuxt Application in the same port?

### DISCUSSION THREAD

**Discussion by wypratama** [nuxt/nuxt#20727] [Questions]
Hi Guys, I started a monorepo project with multiple nuxt 3 apps inside it. Now I want to be able to run all the app with the same port but different base route like first app: '/', and second app: '/blog'. Is there a way to achieve this in Nuxt 3?

**Comment by maou-shonen** [comment]
For your use case try using [Layers](https://nuxt.com/docs/getting-started/layers),
If the logic is more complicated, I suggest you solve it from the reverse proxy server.

**Reply by wypratama** [reply]
I have tried to create a dev layer where it extends from all the app i have.
the first app have this config: 
```ts
export default defineNuxtConfig({
  typescript: {
    shim: false,
    strict: true,
  },
  app: {
    baseURL: '/',
  },
})
```

and the second app have this config: 
```ts
export default defineNuxtConfig({
  typescript: {
    shim: false,
    strict: true,
  },
  app: {
    baseURL: '/blog',
  },
})
```

but everything is merged into / anyway.

**Comment by gzf6** [comment]
Same issue. During development, can I set different baseUrls for multiple apps, but use the same port?

**Comment by Krutie** [comment]
@wypratama Check out this repo: https://github.com/gravitano/nuxt3-tailwind-kit  

This project doesn't use monorepo-approach, but addresses your other requirement well. 

Study the `/apps` folder to see how `/auth`, `/blog`, `/store`, etc can be separate apps, but they all run from the same port because ...each of these apps are authored as a **Nuxt Layer**.  Then it uses `/pages` directory in each of them to create individual app routes such as:
- http://localhost:3000/blog
- http://localhost:3000/store, etc.
